### PR TITLE
Add episode on large file storage with its pros and cons

### LINKED ISSUE #8: Large Files episode write up
Large files episode needs to be written.

Git LFS needs to be included.

Exercise with a binary file would be great.

### PR DESCRIPTION
**Pull request by dokempf**
Feel free to comment and add content!

This fixes #8.

### CODE REVIEW

File: _episodes/12-large-files.md [hunk 1]
```diff
@@ -3,14 +3,164 @@ title: "Large Files"
...
+- "Learning about the `git lfs` commands"
+- "Understanding the disadvantages of `git lfs`"
 keypoints:
-- "First key point. Brief Answer to questions. (FIXME)"
+- "(Large) binary files can grow the repository size immensely and make it unusable"
+- "`git lfs` is an extension that stores large files outside the Git data model"
+- "Use of Git LFS is discouraged in many scenarios."
 ---
 
+TODO: Installation steps for this episode
```
Comment (ivastar) [inline _episodes/12-large-files.md]: Can you please add a link to this set of installation instructions for completeness: https://docs.github.com/en/repositories/working-with-files/managing-large-files/installing-git-large-file-storage

File: _episodes/12-large-files.md [hunk 2]
```diff
@@ -3,14 +3,164 @@ title: "Large Files"
...
+* Some hosting providers - most notably GitHub - apply restrictive quotas to LFS storage.
+  On the free plan, GitHub currently allows 1GB of storage and 1 GB bandwidth per month.
+  As the band width quota counts every single clone by users, **LFS should currently
+  be considered unusable on the GitHub free plan.**
+~~~
+{: .caution}
+
+TODO: Write about `git annex`
+
+TODO: Write about other potential solutions.
```
Comment (ivastar) [inline _episodes/12-large-files.md]: Can you please add the ideas from the slide so we can keep track of them:
TODO: Write about other potential solutions, including ignore them in Git repo, handle versioning somewhere else. 

To completely remove a file from history (for the desperate only):
- Squash out all the commits that contains that file
- git filter-branch
- [BFG repo cleaner](https://rtyley.github.io/bfg-repo-cleaner/)
- Start afresh in a new repo

**Review by ivastar — COMMENTED** [review]
A couple of minor suggestions.

File: _episodes/12-large-files.md [hunk 3]
```diff
@@ -3,14 +3,164 @@ title: "Large Files"
...
+git push origin main
+~~~
+{: .language-bash}
+
+~~~
+Uploading LFS objects: 100% (1/1), 17 B | 0 B/s, done.                          
+~~~
+{: .output}
+
+~~~
```
Comment (rgaiacs) [inline _episodes/12-large-files.md]: Delete this line.

File: _episodes/12-large-files.md [hunk 4]
```diff
@@ -3,14 +3,164 @@ title: "Large Files"
...
+patterns can be passed to `git lfs track`. However, you should be careful to quote
+these patterns, as they might otherwise get expanded by to existing files by your shell.
+For example, tracking all PDFs with LFS could be achieved with the following command:
+
+~~~
+git lfs track "*.pdf"
+~~~
+{: .language-bash}
+~~~
+{: .callout}
```
Comment (rgaiacs) [inline _episodes/12-large-files.md:138]: This is in the wrong place.

**Review by rgaiacs — CHANGES_REQUESTED** [review]
Looks good to me. Minor changes to fix conversion to HTML.

**Comment by mihai-sysbio** [comment]
There is also a small rendering issue, solving in progress.

**Comment by mihai-sysbio** [comment]
Everything above seems to be resolved. The text reads nicely, and is comprehensive yet easy to follow. In my view, ready to merge.

File: _episodes/12-large-files.md (hunk 1 above)
Comment (dokempf) [reply _episodes/12-large-files.md]: Added to the respective issue #26

File: _episodes/12-large-files.md (hunk 2 above)
Comment (dokempf) [reply _episodes/12-large-files.md]: Added to #28 (to avoid TODO notes in the repository)

File: _episodes/12-large-files.md (hunk 4 above)
Comment (dokempf) [reply _episodes/12-large-files.md:138]: Should be fixed by now.

**Comment by dokempf** [comment]
I addressed all comments. TODOs are now documented as issues to make the lesson itself usable as is.

**Comment by ivastar** [comment]
Thank you fro this contribution @dokempf!

**Pull request merged**